### PR TITLE
server: Configure content detection via auth webhook

### LINKED ISSUE #1884: Configure content detection via broadcaster stream auth webhook
See "Broadcaster Stream Auth Webhook" in https://github.com/livepeer/go-livepeer/issues/1881

### PR DESCRIPTION
**Pull request by jailuthra**
**What does this pull request do? Explain your changes. (required)**

Enable the user to configure content detection profile using auth webhook.

**Specific updates (required)**

*See commit history*

**How did you test each of these updates (required)**

**Does this pull request close any open issues?**

Fixes #1884 

**Checklist:**

- [x] Read the [contribution guide](./doc/contributing.md)
- [x] `make` runs successfully
- [x] All tests in `./test.sh` pass
- [x] README and other documentation updated
- [ ] [Pending changelog](./CHANGELOG_PENDING.md) updated

### CODE REVIEW

File: server/mediaserver.go [hunk 1]
```diff
@@ -116,6 +116,17 @@ type authWebhookResponse struct {
...
+	Detection        struct {
+		// Run detection on 1/freq segments
+		Freq                       uint `json:"freq"`
+		SceneClassificationProfile struct {
+			SampleRate uint `json:"sampleRate"`
+			Classes    []struct {
+				ID   int    `json:"id"`
+				Name string `json:"name"`
+			} `json:"classes"`
+		} `json:"sceneClassifcationProfile"`
```
Comment (yondonfu) [inline server/mediaserver.go]:
```suggestion
		} `json:"sceneClassificationProfile"`
```

File: server/mediaserver.go [hunk 2]
```diff
@@ -386,6 +404,25 @@ func jsonProfileToVideoProfile(resp *authWebhookResponse) ([]ffmpeg.VideoProfile
...
+	for _, class := range resp.Detection.SceneClassificationProfile.Classes {
+		sceneClassification.Classes = append(sceneClassification.Classes,
+			ffmpeg.DetectorClass{
+				ID:   class.ID,
+				Name: class.Name,
+			})
+	}
+	detection := core.DetectionConfig{
+		Freq:     resp.Detection.Freq,
+		Profiles: []ffmpeg.DetectorProfile{&sceneClassification},
```
Comment (yondonfu) [inline server/mediaserver.go:424]: Is a pointer to `sceneClassification` required when creating the `[]ffmeg.DetectorProfile` slice or would the struct value be fine as well?

File: server/mediaserver.go [hunk 3]
```diff
@@ -386,6 +404,25 @@ func jsonProfileToVideoProfile(resp *authWebhookResponse) ([]ffmpeg.VideoProfile
 	return profiles, nil
 }
 
+func jsonDetectionToDetectorProfile(resp *authWebhookResponse) core.DetectionConfig {
```
Comment (yondonfu) [inline server/mediaserver.go]:
```suggestion
func jsonDetectionToDetectionConfig(resp *authWebhookResponse) core.DetectionConfig {
```

^ Feels a bit more accurate naming wise given that the return value is `core.DetectionConfig`. WDYT?

**Review by yondonfu — CHANGES_REQUESTED** [review]

File: server/mediaserver.go (hunk 2 above)
Comment (jailuthra) [reply server/mediaserver.go:424]: Currently the `DetectorProfile` interface is implemented for the pointer to the struct `*SceneClassificationProfile` https://github.com/livepeer/lpms/pull/228/files#diff-f3da988682c76502f83c01535ea7a53ae69403b27392275ef770e2bc69ce724aR28

In retrospect we should've implemented the interface directly for the struct type, but changing it now means touching 3 different PRs - are there any obvious cons to using a struct pointer here?
If yes, I don't mind changing the 3 PRs - If no, let's keep this as is for now.

Comment (yondonfu) [reply server/mediaserver.go:424]: Implementing the interface with the pointer seems okay for now. Just wanted to be wary about mixing usage of the struct value and the struct pointer.

**Review by yondonfu — APPROVED** [review]
LGTM after squashing

**Pull request merged**